WordPressEditor/src/androidTest/java/org/wordpress/android/editor/integration/EditorFragmentTest.java

```diff
@@ -0,0 +1,108 @@
+package org.wordpress.android.editor.integration;
+
+import android.app.Activity;
+import android.test.ActivityInstrumentationTestCase2;
+import android.view.View;
+import android.widget.ToggleButton;
+
+import org.wordpress.android.editor.R;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.wordpress.android.editor.TestingUtils.waitFor;
+
+public class EditorFragmentTest extends ActivityInstrumentationTestCase2<MockEditorActivity> {
+    private Activity mActivity;
+    private EditorFragmentForTests mFragment;
+
+    public EditorFragmentTest() {
+        super(MockEditorActivity.class);
+    }
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        mActivity = getActivity();
+        mFragment = (EditorFragmentForTests) mActivity.getFragmentManager().findFragmentByTag("editorFragment");
+    }
+
+    public void testDomLoadedCallbackReceived() {
+        // initJsEditor() should have been called on setup
+        assertTrue(mFragment.mInitCalled);
+
+        long start = System.currentTimeMillis();
+        while(!mFragment.mDomLoaded) {
+            waitFor(10);
```

Comment:
I see we have a few `sleep()` calls in these tests. I get why they are there, however they tend to make tests brittle (esp when running within a CI environment). Not sure if we can do anything about them now...but may be an opportunity for future refactoring.